File: ConsolaProgramas/Formularios/frmErrors.vb
Public Class frmErrors
    Dim Valido As Boolean = True
    Private _isOk As Boolean = False

    Public ReadOnly Property Errores As Integer
        Get
            Return lvErrores.Items.Count
        End Get
    End Property

    Public ReadOnly Property isOk As Boolean
        Get
            Return _isOk
        End Get
    End Property
    Public Sub NuevoError(Mensaje As String, Tipo As Integer)
        If Tipo = 0 Then
            lvErrores.Items.Add("  " & Mensaje, 0)
            Valido = False
        Else
            lvErrores.Items.Add("  " & Mensaje, 1)
        End If
    End Sub
    Private Sub frmErrors_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        If Valido Then
            cmdAceptar.Visible = True
        Else
            cmdAceptar.Visible = False
        End If
    End Sub
    Private Sub cmdCancelar_Click(sender As Object, e As EventArgs) Handles cmdCancelar.Click
        lvErrores.Clear()
        Valido = True
        Close()
    End Sub

    'Private Sub frmErrors_FormClosing(sender As Object, e As Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
    '    lvErrores.Clear()
    '    Valido = True
    'End Sub

    Private Sub cmdAceptar_Click(sender As Object, e As EventArgs) Handles cmdAceptar.Click
        _isOk = True
        Close()
    End Sub
End Class

File: ConsolaProgramasCS/Shared Functions/frmErrors.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace ConsolaProgramasCS.Shared_Functions
{
    public partial class frmErrors : DevExpress.XtraEditors.XtraForm
    {
        bool Valido = true;
        private bool _isOk = false;

        public int Errores { get { return lvErrores.Items.Count; } }
        public bool isOk { get { return _isOk; } }
        
        public frmErrors()
        {
            InitializeComponent();
        }

        private void frmErrors_Load(object sender, EventArgs e)
        {
            if(Valido)
            {
                cmdAceptar.Visible = true;
            }
            else
            {
                cmdAceptar.Visible = false;
            }
      
        }
        public void NuevoError(string mensaje, int tipo)
        {
            if (tipo == 0)
            {
                lvErrores.Items.Add("  " + mensaje, 0);
                Valido = false;
            }
            else
            {
                lvErrores.Items.Add("  " + mensaje, 1);
            }
         
        }

        private void cmdCancelar_Click(object sender, EventArgs e)
        {
            lvErrores.Clear();
            Valido = true;
            Close();
        }

        private void cmdAceptar_Click(object sender, EventArgs e)
        {
            _isOk = true;
            Close();
        }
    }
}